src/header.rs

```diff
@@ -0,0 +1,101 @@
+use carbonado::error::CarbonadoError;
+use log::debug;
+use serde::{Deserialize, Serialize};
+use serde_cbor;
+use serde_cbor::Error;
+
+// use crate::{
+//     constants::{Format, MAGICNO},
+//     error::CarbonadoError,
+// };
+
+/// Contains deserialized copies of the data kept in the Catalog header.
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct CatHeader {
+    /// Number of bytes added to pad to align zfec chunks and bao slices.
+    pub cbor_len: u8,
+    /// Bytes that are normally zero but can contain extra data if needed
+    pub metadata: Option<Vec<u8>>,
+}
+
+impl CatHeader {
+    /// 160 bytes should be added for Carbonado headers.
+    pub fn len() -> usize {
+        12 + 33 + 32 + 64 + 1 + 1 + 4 + 4 + 8 + 1
+    }
+
+    /// Creates a new Catalog Header struct using the provided parameters.
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(metadata: Vec<u8>) -> Result<Self, Error> {
+        //type Error = CarbonadoError;
+        let cbor_data = serde_cbor::to_vec(&metadata)?;
+        println!("cbor data; {:?}", cbor_data);
+
+        let cbor_len = cbor_data.len() as u8;
+
+        println!("cbor length ; {:?}", cbor_len);
+
+        let magic_no = carbonado::constants::MAGICNO;
```

Comment:
Cat files will use their own magic number, remember what I said:

Use this for magic no:
https://github.com/diba-io/carbonado/blob/518188593fc2a86bf2fa687f9165b83904c0d64f/src/constants.rs#L5
Except use:
`CARBONCAT00` instead of `CARBONADO00`